src/main/java/org/killbill/billing/plugin/analytics/http/AnalyticsAccountResource.java

```diff
@@ -86,4 +87,15 @@ public Result doPut(@Named("accountId") final UUID accountId,
             return Results.with(new ExceptionResponse(e, true), Status.SERVER_ERROR);
         }
     }
+
+    @PUT
+    @Path("/")
+    public Result doPut(@Header(HDR_CREATED_BY) final Optional<String> createdBy,
+                        @Header(HDR_REASON) final Optional<String> reason,
+                        @Header(HDR_COMMENT) final Optional<String> comment,
+                        @Local @Named("killbill_tenant") final Tenant tenant) {
+        final CallContext context = createCallContext(createdBy, reason, comment, null, tenant);
+        analyticsUserApi.rebuildAnalyticsForAllAccounts(context);
+        return Results.with(Status.NO_CONTENT);
+    }
 }
```

Comment:
This endpoint is not asynchronous yet. Could you point me to some similar code so I can take a look and implement the same here?

Reply:
Good point. For large datasets, clients would timeout.

An easy and elegant way to tackle this would be to simply insert a _job_ in the notification queue, and the plugin will process it asynchronously (and reliably). Check https://github.com/killbill/killbill-analytics-plugin/blob/4bc554af7d0a0f870d4fd0b9596a649c3c6a265c/src/main/java/org/killbill/billing/plugin/analytics/AnalyticsListener.java#L230

Reply:
1. Yeah, I think it's fine for now. 👍 
2. We actually want all of these notifications. That way, all Kill Bill servers can process refreshes in parallel, as fast as they can.